hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/api/NodeEndpoint.java

```diff
@@ -171,4 +181,163 @@ private DatanodeStorageReport getStorageReport(DatanodeDetails datanode) {
+    try {
+      for (String uuid : uuids) {
+        DatanodeDetails nodeByUuid = nodeManager.getNodeByUuid(uuid);
+        try {
+          if (preChecksSuccess(nodeByUuid)) {
+            removedDatanodes.add(DatanodeMetadata.newBuilder()
+                .withHostname(nodeManager.getHostName(nodeByUuid))
+                .withUUid(uuid)
+                .withState(nodeManager.getNodeStatus(nodeByUuid).getHealth())
+                .build());
+            nodeManager.removeNode(nodeByUuid);
+          } else {
+            Response.ResponseBuilder builder = Response.status(Response.Status.BAD_REQUEST);
+            builder.entity("{\n" +
+                "    \"Invalid request: Pre-checks failed for selected datanodes. DataNode should pass following " +
+                "pre-checks.\": [\n" +
+                "        {\n" +
+                "            \"title\": \"Incorrect State\",\n" +
+                "            \"description\": \"DataNode should be in either DECOMMISSIONED operational state or " +
+                "DEAD node state.\"\n" +
+                "        },\n" +
+                "        {\n" +
+                "            \"title\": \"Open Containers\",\n" +
+                "            \"description\": \"Containers are open for few or all selected datanodes.\"\n" +
+                "        },\n" +
+                "        {\n" +
+                "            \"title\": \"Open Pipelines\",\n" +
+                "            \"description\": \"Pipelines are open for few or all selected datanodes.\"\n" +
+                "        }\n" +
+                "    ]\n" +
+                "}");
+            failedDatanodes.add(DatanodeMetadata.newBuilder()
+                .withHostname(nodeManager.getHostName(nodeByUuid))
+                .withUUid(uuid)
+                .withOperationalState(nodeByUuid.getPersistedOpState())
+                .withState(nodeManager.getNodeStatus(nodeByUuid).getHealth())
+                .build());
+            // Build and return the response
+            return builder.build();
+          }
+        } catch (NodeNotFoundException nnfe) {
+          LOG.error("Selected node {} not found : {} ", uuid, nnfe);
+          notFoundDatanodes.add(DatanodeMetadata.newBuilder()
+                  .withHostname("")
+                  .withState(NodeState.DEAD)
+              .withUUid(uuid).build());
+        }
+      }
+    } catch (Exception exp) {
+      LOG.error("Unexpected Error while removing datanodes : {} ", exp);
+      throw new WebApplicationException(exp, Response.Status.INTERNAL_SERVER_ERROR);
+    }
+
+    RemoveDataNodesResponseWrapper removeDataNodesResponseWrapper = new RemoveDataNodesResponseWrapper();
+
+    if (!failedDatanodes.isEmpty()) {
+      DatanodesResponse failedNodesResp =
+          new DatanodesResponse(failedDatanodes.size(), failedDatanodes);
+      failedNodesResp.setMessage("Invalid request: Nodes should be in either DECOMMISSIONED or " +
```

Comment:
description is not correct, it can have open pipeline and open container.

Reply:
every failed node can have different reason of failure

Reply:
Ok done. This is how response will be now:

```
{
    "datanodesResponseMap": {
        "failedDatanodes": {
            "totalCount": 1,
            "errors": {
                "28607ed0-181e-4adc-9d60-08591ec53ccd": "DataNode should be in either DECOMMISSIONED operational state or DEAD node state."
            }
        },
        "notFoundDatanodes": {
            "totalCount": 1,
            "datanodes": [
                {
                    "uuid": "de754748-5a61-45ce-a92a-532c70b0ab2b",
                    "hostname": "",
                    "state": "DEAD",
                    "pipelines": null
                }
            ]
        },
        "removedDatanodes": {
            "totalCount": 1,
            "datanodes": [
                {
                    "uuid": "04bea959-ed7b-4e9f-8c06-8e2dc22701fe",
                    "hostname": "ozone-ha-datanode-1.ozone-ha_default",
                    "state": "HEALTHY",
                    "pipelines": null
                }
            ]
        }
    }
}
```